homeassistant/components/livisi/config_flow.py

```diff
@@ -25,48 +24,42 @@ def __init__(self) -> None:
         self._auth_headers: dict[str, Any] = {}
         self._aio_livisi: AioLivisi = None
         self._access_data = None
+        self.data_schema = vol.Schema(
+            {
+                vol.Required(CONF_HOST): str,
+                vol.Required(CONF_PASSWORD): str,
+            }
+        )
 
-    async def async_step_user(self, user_input=None) -> FlowResult:
-        """Handle the initial step."""
-        if user_input is not None:
-            host: str = user_input[CONF_HOST]
-            self._host = host
-            return await self.async_step_credentials(None)
-
-        return self.async_show_form(step_id="user", data_schema=HOST_SCHEMA)
-
-    async def async_step_credentials(
-        self, user_input: dict[str, Any] = None
+    async def async_step_user(
+        self, user_input: dict[str, str] | None = None
     ) -> FlowResult:
-        """Handle the credentials step."""
-        found_errors: dict[str, str] = {}
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
         if user_input is not None:
             try:
-                return await self._login(user_input)
-            except errors.WrongCredentialException:
-                found_errors["base"] = "wrong_password"
-            except errors.ShcUnreachableException:
-                found_errors["base"] = "shc_unreachable"
-            except errors.IncorrectIpAddressException:
-                found_errors["base"] = "wrong_ip_address"
-        else:
-            user_input = {}
-
-        schema = vol.Schema(
-            {
-                vol.Required(CONF_PASSWORD, default=user_input.get(CONF_PASSWORD)): str,
-            }
-        )
+                ip_password: dict[str, str] = user_input
+                return await self.async_step_credentials(ip_password)
+            except livisi_errors.WrongCredentialException:
+                errors["base"] = "wrong_password"
+            except livisi_errors.ShcUnreachableException:
+                errors["base"] = "shc_unreachable"
+            except livisi_errors.IncorrectIpAddressException:
+                errors["base"] = "wrong_ip_address"
 
         return self.async_show_form(
-            step_id="credentials", data_schema=schema, errors=found_errors
+            step_id="credentials", data_schema=self.data_schema, errors=errors
         )
 
-    async def _login(self, user_input: dict[str, Any]):
+    async def async_step_credentials(self, user_input: dict[str, str]) -> FlowResult:
```

Comment:
Please remove this step. It's not needed. Just call `_login` from the user step.